std/algorithm/searching.d

```diff
@@ -1361,7 +1361,19 @@ if (isInputRange!InputRange &&
 
     alias EType  = ElementType!R;
 
-    static if (isNarrowString!R)
+    // If the haystack is a SortedRange we can use binary search to find the needle.
+    // Works only for the default find predicate and any SortedRange predicate.
+    // 8829 enhancement
+    import std.range: SortedRange;
+    static if (is(InputRange : SortedRange!TT, TT) && isDefaultPred)
+    {
+        auto lb = haystack.lowerBound(needle);
+        if (lb.length == 0 || lb.length == haystack.length || haystack[lb.length] != needle)
```

Comment:
Why do you have `lb.length == 0` there? It means there is no element less than the needle in the haystack, so the haystack may actually start with the needle. No?